src/main/kotlin/com/infomaniak/core/extensions/MutableSetExt.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Infomaniak Core - Android
+ * Copyright (C) 2025 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.core.extensions
+
+import android.os.Build
+
+fun MutableSet<String>.addPermission(
+    permission: String,
+    minSdk: Int = minSdkVersion,
+    maxSdk: Int = Build.VERSION_CODES.CUR_DEVELOPMENT,
+) {
+    if (Build.VERSION.SDK_INT in minSdk..<maxSdk) add(permission)
```

Comment:
Why are you excluding the maxSdk version ?

Reply:
Because we always talk and know when a permission got "excluded", not the last version it's on
It's always : "this permission is needed until ..."